### PR TITLE
fix(build): change if condition to faill over no credentials endpoint

### PR DESCRIPTION
**Pull request by smothiki**

### CODE REVIEW

File: manifests/deis-slugbuilder.yaml [hunk 1]
```diff
@@ -11,14 +11,14 @@ spec:
...
         - name: DEBUG
           value: "1"
         - name: TAR_URL
           value: "https://s3.amazonaws.com/mydaffa/mytar.tgz"
         - name : put_url
-          value: "https://s3-us-west-1.amazonaws.com/myjaffa"
+          value: "https://s3-us-west-1.amazonaws.com/myjaffa/tmp"
       volumeMounts:
         - name: object-store
           mountPath: /var/run/secrets/object/store
```
Comment (arschles) [inline manifests/deis-slugbuilder.yaml:24]: @smothiki looks like this file is just for testing. right?

Comment (smothiki) [inline manifests/deis-slugbuilder.yaml:24]: Yup. you can change the kube secrets and URL endpoints

Comment (arschles) [inline manifests/deis-slugbuilder.yaml:24]: cool, thanks

File: rootfs/builder/build.sh [hunk 2]
```diff
@@ -202,8 +202,8 @@ if [[ "$slug_file" != "-" ]]; then
 					mc --quiet config host add $domain $keyID $secretKey
 					mc --quiet cp $slug_file $put_url/
 				fi
-			fi
-		else
-      curl -0 -s -o /dev/null -X PUT -T $slug_file "$put_url"
-  	fi
+			else
+      	curl -0 -s -o /dev/null -X PUT -T $slug_file "$put_url"
```
Comment (arschles) [inline rootfs/builder/build.sh]: @smothiki should the `curl` be indented so that it's 'under' the `else`?

Comment (smothiki) [inline rootfs/builder/build.sh]: My bad Its under else but the indentation was wrong. Some how indendation got messed up. I'll fix it ASAP

Comment (arschles) [inline rootfs/builder/build.sh]: k, thanks

**Comment by arschles** [comment]
@bacongobbler can you review?

File: rootfs/builder/build.sh [hunk 3]
```diff
@@ -25,10 +25,10 @@ if ! [[ -z "${TAR_URL}" ]]; then
 			tar -xzf /tmp/slug.tgz -C /app/
 			unset TAR_URL
 		fi
+	else
+		curl -s "$TAR_URL" | tar -xzC /app/
+		unset TAR_URL
 	fi
-else
```
Comment (bacongobbler) [inline rootfs/builder/build.sh]: i think we still want this logic here. if $TAR_URL is present but the secret is not, we should still pull the tarball.

File: rootfs/builder/build.sh [hunk 4]
```diff
@@ -202,8 +202,8 @@ if [[ "$slug_file" != "-" ]]; then
 					mc --quiet config host add $domain $keyID $secretKey
 					mc --quiet cp $slug_file $put_url/
 				fi
+			else
+				curl -0 -s -o /dev/null -X PUT -T $slug_file "$put_url"
 			fi
-		else
```
Comment (bacongobbler) [inline rootfs/builder/build.sh]: same here. We still want this logic to happen if the secret does not exist.

**Pull request merged**